### PR TITLE
odoo15: init at 15.0-20230720

### PR DESCRIPTION
**Pull request by karolinschlegel**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] ~Tested, as applicable:~
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] ~Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)~
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by karolinschlegel** [comment]
Hi @mkg20001 ! You may also want to take a look at this.

File: pkgs/applications/finance/odoo/odoo15.nix [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    xlwt
+    zeep
+  ];
+
+  # takes 5+ minutes and there are not files to strip
+  dontStrip = true;
+
+  passthru = {
+    updateScript = ./update.sh;
+    tests = { inherit (nixosTests) odoo; };
```
Comment (mkg20001) [inline pkgs/applications/finance/odoo/odoo15.nix:119]: this uses odoo, not odoo15. needs to be adjusted or removed.

Comment (karolinschlegel) [reply pkgs/applications/finance/odoo/odoo15.nix:119]: I extended the test to make the package configurable. Do you have an idea how to pass the attribute to the test?

Comment (mkg20001) [reply pkgs/applications/finance/odoo/odoo15.nix:119]: you could make it `package ? "odoo"` and then later use `package = pkgs.${package}`

Comment (mkg20001) [reply pkgs/applications/finance/odoo/odoo15.nix:119]: but you could also add odoo15 in the top of all-tests as it's just a file loaded by callPackage aswell

Comment (karolinschlegel) [reply pkgs/applications/finance/odoo/odoo15.nix:119]: > edit: theres with pkgs at the top so odoo15 should just work

`tests = { inherit (nixosTests) odoo; };` does not work. 
So I need something like
`tests = { inherit (nixosTests) odoo { package = pkgs.odoo15}; };` but I'm not sure how to achieve that.

> but you could also add odoo15 in the top of all-tests as it's just a file loaded by callPackage aswell

Probably that's the way to go?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -6091,6 +6091,7 @@ with pkgs;
   };
 
   odoo = callPackage ../applications/finance/odoo { };
+  odoo15 = callPackage ../applications/finance/odoo/odoo15.nix { };
```
Comment (mkg20001) [inline pkgs/top-level/all-packages.nix:6094]:
```suggestion
  odoo15 = callPackage ../applications/finance/odoo/odoo15.nix { package = odoo15; };
```

File: pkgs/applications/finance/odoo/odoo15.nix (hunk 1 above)
Comment (mkg20001) [reply pkgs/applications/finance/odoo/odoo15.nix:119]:
```suggestion
    tests = { inherit (nixosTests) odoo15; };
```

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (karolinschlegel) [reply pkgs/top-level/all-packages.nix:6094]: I added it to `all-tests.nix`.

File: pkgs/applications/finance/odoo/odoo15.nix (hunk 1 above)
Comment (karolinschlegel) [reply pkgs/applications/finance/odoo/odoo15.nix]: Thank you, done!

**Pull request merged**